scrape/scrape_test.go

```diff
@@ -660,42 +661,109 @@ func TestScrapeLoopRunCreatesStaleMarkersOnParseFailure(t *testing.T) {
 }
 
 func TestScrapeLoopAppend(t *testing.T) {
-	app := &collectResultAppender{}
+	tests := []struct {
+		title           string
+		honorLabels     bool
+		scrapeLabels    string
+		discoveryLabels []string
+		expected        []string
+	}{
+		{
+			// When "honor_labels" is not set
+			// label name collision is handler by adding a prefix.
+			title:           "Label name collision",
+			honorLabels:     false,
+			scrapeLabels:    `metric{n="1"} 0`,
+			discoveryLabels: []string{"n", "2"},
+			expected:        []string{"metric", "0", "exported_n", "1", "n", "2"},
```

Comment:
Mh, so this encodes the metric name in the first entry, then a value in the 2nd and then label pairs in the subsequent one? Seems rather convoluted and I had to read the test code to check what the test case actually expresses.

Maybe split it up into dedicated fields?

Reply:
now when I see it changed it looks so logical and so simple , thanks for the :1st_place_medal:  comments